### DISCUSSION TITLE
is it possible to use rdkit with GPU accelaration

### DISCUSSION THREAD

**Discussion by debanjansen48** [rdkit/rdkit#5594] [Ideas]
I am interested to prepare a ultra large 3D chemical library for ultra large virtual screening with gpu acceleration. The crucial step is ligand preparation. RDkit is a beautiful tool to convert smi file list to 3D optimized structure. CPU based such job is too much time taking. Is it possible to apply GPU acceleration with Rdkit for 3D ligand preparation, minimization?

**Comment by kienerj** [comment]
The short answer is no as in rdkit does not offer algorithms that run specifically on GPUs.

it is however an interesting discussion in a more general term. Before thinking about GPUs I would first think or look about how optimized the specific algorithms are for the CPU. Do they use modern vector instructions, eg AVX? Can the algorithms even use them or take benefit from them? Are there other potentially "easy" wins to increase performance?

Same for GPU, most algorithms simply can't be applied to a gpu or they won't profit from it (in fact run much slower). 

And before all that, you can simply parallelize the workload over many, many cpus either with something simple as multiprocessing or joblib or by the sounds of it run it on a spark cluster or similar (with Pyspark). In fact I have experimented with Pyspark and rdkit and it works. All you need is the cluster (which you can do in cloud but at a cost obviously).

**Comment by debanjansen48** [comment]
Thanks for your kind words.... sir affording many more cpu is less economic
frandldy than having  a minimal Gpu. I am not a native English speaker.
Don't mistake me. I felt gpu based processing may facilitate to proess
large ligs in minimal time...
